File: huaweicloud-cs-java-examples/src/main/java/com/huaweicloud/cs/java/model/HBaseInputFormat.java
package com.huaweicloud.cs.java.model;

import org.apache.flink.addons.hbase.TableInputFormat;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.hadoop.hbase.client.Result;
import org.apache.hadoop.hbase.client.Scan;
import org.apache.hadoop.hbase.util.Bytes;

/**
 * This class implements an TableInputFormat for HBase.
 */
public class HBaseInputFormat extends TableInputFormat<Tuple2<String, String>> {

    private Tuple2<String, String> reuse = new Tuple2<>();

    @Override
    public String getTableName() {
        return FlinkHBaseConstants.TEST_TABLE_NAME;
    }

    @Override
    protected Scan getScanner() {
        Scan scan = new Scan();
        scan.addColumn(FlinkHBaseConstants.CF_SOME, FlinkHBaseConstants.Q_SOME);
        return scan;
    }

    @Override
    protected Tuple2<String, String> mapResultToTuple(Result r) {
        String key = Bytes.toString(r.getRow());
        String val = Bytes.toString(r.getValue(FlinkHBaseConstants.CF_SOME, FlinkHBaseConstants.Q_SOME));
        reuse.setField(key, 0);
        reuse.setField(val, 1);
        return reuse;
    }

}

File: huaweicloud-cs-examples/src/main/scala/com/huaweicloud/cs/scala/model/HBaseInputFormat.scala
package com.huaweicloud.cs.scala.model

import org.apache.flink.addons.hbase.TableInputFormat
import org.apache.flink.api.java.tuple.Tuple2
import org.apache.hadoop.hbase.client.{Result, Scan}
import org.apache.hadoop.hbase.util.Bytes

/**
  * This class implements an TableInputFormat for HBase.
  */
class HBaseInputFormat extends TableInputFormat[Tuple2[String, String]] {

  val reuse = new Tuple2[String, String]

  def getTableName = FlinkHbaseConstants.TEST_TABLE_NAME

  def getScanner: Scan = {
    val scan = new Scan
    scan.addColumn(FlinkHbaseConstants.CF_SOME, FlinkHbaseConstants.Q_SOME)
    scan
  }

  def mapResultToTuple(r: Result): Tuple2[String, String] = {
    val key = Bytes.toString(r.getRow)
    val value = Bytes.toString(r.getValue(FlinkHbaseConstants.CF_SOME, FlinkHbaseConstants.Q_SOME))
    reuse.setField(key, 0)
    reuse.setField(value, 1)
    reuse
  }

}